### ISSUE TITLE
[Bug]: there might be differences between fsdp_wrapper_model and non-wrappered model

### ISSUE THREAD

**Issue by lucywang720** [RLinf/RLinf#660]
### Affected component

模型输出的值会被fsdp或不fsdp影响

### Bug description

我在用你们的库训我自定义的qwen模型（qwen2_5_vl，最后decode改为sigmoid head）
    def init_worker(self):
        """初始化 FSDP 模型和处理器"""
        # 1. 设置模型并包装为 FSDP (内部会调用 model_provider_func)
        # self.setup_model_and_optimizer()
        self.setup_model()
        
        # 2. 初始化处理器
        self.processor = AutoProcessor.from_pretrained(
            self.cfg.reward.tokenizer.tokenizer_model
        )
        
        print(self.processor.image_processor.merge_size)
        
        self.merge_size = getattr(self.processor.image_processor, "merge_size", 2)
        
        # 3. 如果开启了卸载，将初始化后的权重移至 CPU
        if self.enable_offload:
            self.offload_param_and_grad()
        
        self.log_info(f"FSDP Reward Worker initialized at rank {self._rank}")

    def model_provider_func(self) -> torch.nn.Module:
        """重写此函数以指定如何加载原始模型"""
        # 注意：不要在这里使用 device_map="auto"
        # 模型会先加载到 CPU，然后由 FSDP 进行分片
        model = Custom_Qwen2_5_VLForConditionalGeneration.from_pretrained(
            self.cfg.reward.model.model_path,
            torch_dtype=torch.bfloat16,
        )
        # print(model.)
        model.eval()
        return model

现在的问题是我发现在setup model的时候
        # self.model = self._strategy.wrap_model(
        #     model=module, device_mesh=self._device_mesh
        # )
        self.model=module

是否使用self._strategy.wrap_model会对结果造成非常大的影响（对同一个输入，大约输出在0.3027 vs 0.3145）这种0.01的量级上。
我理解fsdp相比于非fsdp只是会多一个数据传输的流程，为什么会差别这么大呢？谢谢

### Configuration YAML file

YAML config file:

You can also paste the full config here.
```
<paste YAML content here>
```

### Log file

You can find the log file in `logs/` folder or the $output_dir/$experiment_name folder (defined in the yaml config) if you are using our example scripts.

Log file:

If you cannot find the log, please provide the full log messages here.
```
<paste log here>
```

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).
